src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -335,7 +335,14 @@ rpmostree_sysroot_upgrader_new (OstreeSysroot              *sysroot,
 RpmOstreeOrigin *
 rpmostree_sysroot_upgrader_dup_origin (RpmOstreeSysrootUpgrader *self)
 {
-  return rpmostree_origin_dup (self->origin);
+  g_autoptr(RpmOstreeOrigin) ret = rpmostree_origin_dup (self->origin);
+  /* A bit hacky; here we clean out the live state which is deployment specific.
+   * The callers of this API all want a dup so they can create a new one based
+   * on it. Perhaps we should rename to
+   * rpmostree_sysroot_upgrader_prepare_new_origin()?
+   */
+  rpmostree_origin_set_live_state (ret, NULL, NULL);
```

Comment:
Maybe all such "live" options that we don't want to carry over should be in a common origin section, e.g. `[live]`. And then the sysroot upgrader can just do `g_key_file_remove_group ()`.